### DISCUSSION TITLE
FR: Support external vector store

### DISCUSSION THREAD

**Discussion by corani** [logancyang/obsidian-copilot#1166] [Q&A]
My vault has about 5000 notes collected over the years, some of which are quite large. It also contains PDF documents, which would increase the size even more once they can be indexed. I'm concerned that the in-memory vector store won't be able to handle this kind of volume, as Obsidian by itself already struggles a bit at times.

Would it be possible to support some external (local) vector stores like Weaviate, Chroma or Milvus?

I'm already using Ollama locally for the LLM and embeddings, running an additional vector store wouldn't be an issue, and all of them have a REST API so it should be possible to use them from Obsidian.

**Comment by logancyang** [comment]
As long as it doesn't require the user to manually spin up another local server, of course. You are welcome to contribute if you think it's doable. My only priority is to make the UX as simple as possible.
